### PR TITLE
stopping all arm actions

### PR DESCRIPTION
**Pull request by lanseafood**
## Linked Issues:
| Jira Ticket 🎟️   | [Oceanwater-553](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-553) |
| ----------- | ----------- |
| EPIC ⚡| [Oceanwater-551](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-551) |
| Github :octocat:  | #66 (parent) |

## Summary of Changes
* goal: updating arm actions s.t. a faults trigger will halt arm movement
* updating trajectory async executor to keep track of arm fault boolean 
* updating arm actions to listen to faults
  * stow
  * unstow
  * Move guarded
  * grind
  * dig circular
  * dig linear
  * deliver sample

## Test
* `roslaunch ow europa_terminator_workspace.launch`
* in new terminal, try any of the following actions
python unstow_action_client.py
python stow_action_client.py
python guarded_move_action_client.py
python grind_action_client.py
python dig_linear_action_client.py
python dig_circular_action_client.py
python deliver_action_client.py
* in middle of movement, turn faults on
* `rosrun dynamic_reconfigure dynparam set /faults prox_pitch_encoder_failure True`
* observe gazebo, and see arm stop movement

### CODE REVIEW

**Comment by Samahu** [comment]
I added anjan to the PR

**Comment by Samahu** [comment]
@lanseafood Note that this merges into Anjan's PR. If Anjan merges his branch tomorrow and this PR not merged before then we need to change the target of this PR merge to **melodic-devel**

File: ow_lander/scripts/trajectory_async_execution.py
Comment (Samahu) [inline ow_lander/scripts/trajectory_async_execution.py:45]:
```suggestion
    return not self.arm_fault
```

File: ow_lander/scripts/deliver_action_server.py
Comment (Samahu) [inline ow_lander/scripts/deliver_action_server.py:89]:
```suggestion
        success = self.trajectory_async_executer.success() and self.trajectory_async_executer.wait()
```

File: ow_lander/scripts/dig_circular_action_server.py
Comment (Samahu) [inline ow_lander/scripts/dig_circular_action_server.py:115]:
```suggestion
        success = self.trajectory_async_executer.success() and self.trajectory_async_executer.wait()
```

File: ow_lander/scripts/guarded_move_action_server.py
Comment (Samahu) [inline ow_lander/scripts/guarded_move_action_server.py:45]: Unused
```suggestion
```

**Review by Samahu — APPROVED** [review]
Serves the purpose.

I have approved the PR but there are several cases where you used the binary **& operator** but I think you meant to use the python **and** + other minor improvements to the code.

File: ow_lander/scripts/dig_circular_action_server.py
Comment (lanseafood) [reply ow_lander/scripts/dig_circular_action_server.py:115]: oops yea been doing cpp for too long lol, will adjust!

**Comment by lanseafood** [comment]
> @lanseafood Note that this merges into Anjan's PR. If Anjan merges his branch tomorrow and this PR not merged before then we need to change the target of this PR merge to **melodic-devel**

yea, I'll wait for 2 approvals before merging. and if Anjan's is merged first I'll just move it to melodic devel.

**Review by anjan1984 — APPROVED** [review]
checked the arm operations

**Review by kmdalal — APPROVED** [review]

**Pull request merged**
